### PR TITLE
[TranslatorBundle] use the common export logic instead of a custom export

### PR DESCRIPTION
**Pull request by Numkil**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | 

[TranslatorBundle] use the common export logic instead of a custom export but only from the settings view and not from commandline.

### CODE REVIEW

File: src/Kunstmaan/TranslatorBundle/AdminList/TranslationAdminListConfigurator.php [hunk 1]
```diff
@@ -28,6 +30,11 @@ class TranslationAdminListConfigurator extends AbstractDoctrineDBALAdminListConf
      */
     protected $locale;
 
+    /**
+     * @var Field[]
+     */
+    private $exportFields = array();
```
Comment (acrobat) [inline src/Kunstmaan/TranslatorBundle/AdminList/TranslationAdminListConfigurator.php]: Change this to the short array syntax

File: src/Kunstmaan/TranslatorBundle/AdminList/TranslationAdminListConfigurator.php [hunk 2]
```diff
@@ -64,6 +71,30 @@ public function buildFields()
...
+                $this->addExportField($locale, strtoupper($locale));
+            }
+        }
+
+        return $this->exportFields;
+    }
+
+    public function addExportField($name, $header)
+    {
+        $this->exportFields[] = new Field($name, $header, false, null, null);
```
Comment (acrobat) [inline src/Kunstmaan/TranslatorBundle/AdminList/TranslationAdminListConfigurator.php]: The `false, null, null` arguments can be removed as these are the default values

File: src/Kunstmaan/TranslatorBundle/AdminList/TranslationAdminListConfigurator.php [hunk 3]
```diff
@@ -92,6 +123,11 @@ public function canEditInline($item)
         return true;
     }
```
Comment (acrobat) [inline src/Kunstmaan/TranslatorBundle/AdminList/TranslationAdminListConfigurator.php]:
```
/**
 * @return bool
 */
```

File: src/Kunstmaan/TranslatorBundle/Controller/TranslatorCommandController.php [hunk 4]
```diff
@@ -74,6 +74,8 @@ public function importForcedAction()
      */
     public function exportAction()
     {
+        // NEXT_MAJOR this route will be removed because we are now using the common export logic
+        @trigger_error('Calling this route is deprecated since KunstmaanNodeBundle 5.1 and will be removed in KunstmaanNodeBundle 6.0. Use the new route in TranslatorController that uses common export logic.', E_USER_DEPRECATED);
```
Comment (acrobat) [inline src/Kunstmaan/TranslatorBundle/Controller/TranslatorCommandController.php]:
```suggestion
        @trigger_error('Using the "KunstmaanTranslatorBundle_command_export" route is deprecated since KunstmaanTranslatorBundle 5.4 and will be removed in KunstmaanTranslatorBundle 6.0. Use the new route in TranslatorController that uses common export logic.', E_USER_DEPRECATED);
```


I don't understand the `Use the new route in TranslatorController that uses common export logic.` part, is there a replacement action? Or is it just the default adminlist export you mean? Otherwise I would reword this to `Use the default adminlist bundle export functionality instead`

File: src/Kunstmaan/TranslatorBundle/Resources/views/Translator/inline_edit.html.twig [hunk 5]
```diff
@@ -1,6 +1,8 @@
-{% if adminlist.configurator.canEditInline(row) %}
+{% spaceless %}
```
Comment (acrobat) [inline src/Kunstmaan/TranslatorBundle/Resources/views/Translator/inline_edit.html.twig]: The `spaceless` tag is deprecated, use `{% apply spaceless%}{% endapply%}` instead

File: src/Kunstmaan/TranslatorBundle/Controller/TranslatorCommandController.php [hunk 6]
```diff
@@ -74,6 +74,8 @@ public function importForcedAction()
      */
     public function exportAction()
     {
+        // NEXT_MAJOR this route will be removed because we are now using the common export logic
+        @trigger_error('Calling this route is deprecated since KunstmaanNodeBundle 5.1 and will be removed in KunstmaanNodeBundle 6.0. Use the default adminlist bundle export functionality instead.', E_USER_DEPRECATED);
```
Comment (acrobat) [inline src/Kunstmaan/TranslatorBundle/Controller/TranslatorCommandController.php]:
```suggestion
        @trigger_error('Using the "KunstmaanTranslatorBundle_command_export" route is deprecated since KunstmaanTranslatorBundle 5.4 and will be removed in KunstmaanTranslatorBundle 6.0. Use the default adminlist bundle export functionality instead.', E_USER_DEPRECATED);
```

File: src/Kunstmaan/TranslatorBundle/Controller/TranslatorCommandController.php [hunk 7]
```diff
@@ -74,6 +74,8 @@ public function importForcedAction()
      */
```
Comment (acrobat) [inline src/Kunstmaan/TranslatorBundle/Controller/TranslatorCommandController.php]: Also add a `@deprecated` docblock with the same text as the deprecation for IDE completion

File: src/Kunstmaan/TranslatorBundle/Controller/TranslatorCommandController.php [hunk 8]
```diff
@@ -71,9 +71,12 @@ public function importForcedAction()
 
     /**
      * @Route("/export", name="KunstmaanTranslatorBundle_command_export")
+     * @Deprecated Using the "KunstmaanTranslatorBundle_command_export" route is deprecated since KunstmaanTranslatorBundle 5.4 and will be removed in KunstmaanTranslatorBundle 6.0. Use the default adminlist bundle export functionality instead.
```
Comment (acrobat) [inline src/Kunstmaan/TranslatorBundle/Controller/TranslatorCommandController.php]:
```suggestion
     * @deprecated Using the "KunstmaanTranslatorBundle_command_export" route is deprecated since KunstmaanTranslatorBundle 5.4 and will be removed in KunstmaanTranslatorBundle 6.0. Use the default adminlist bundle export functionality instead.
```

`@deprecated` should be all lowercase otherwise this error is triggered

```
[Semantical Error] The annotation "@Deprecated" in method Kunstmaan\TranslatorBundle\Controller\TranslatorCommandController::exportAction() was never imported. Did you maybe forget to add a "use" statement for this annotation?
```

**Comment by acrobat** [comment]
@Numkil Can you check the import from spreadsheet with the exported files. As I get some errors I  think are related to the column headers being a different case (eg nl vs NL).

**Comment by Numkil** [comment]
@acrobat 

I tested with a clean and an old project both using master version of the bundles.
If the managed locales in the config is the same as the locales in the headers. (The importer is case insensitive so doesnt matter if in headers = NL and in managed locales = nl). Everything works perfectly. 

With the clean project i only had to remove 'de' from the managed locales because that was not in my test file. (or add de to headers in excel sheet), after that it imported everything correctly.

**Comment by acrobat** [comment]
@Numkil Can you fix the small code style issue reported by styleci? I will also retest my setup, could be possible the error is on my side.
